src/main/java/io/confluent/connect/elasticsearch/jest/actions/PortableJestCreateIndexBuilder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.connect.elasticsearch.jest.actions;
+
+import io.confluent.connect.elasticsearch.ElasticsearchClient.Version;
+import io.searchbox.indices.CreateIndex;
+
+/**
+ * Portable Jest action builder, across ES versions, to create indexes.
+ * This builder add support for ES version 7 by keeping the type support still enabled, this is
+ * done by passing the include_type_name parameter. This parameter is no longer required with ES 8,
+ * as types should not be used anymore by the time of ES 8 release.
+ */
+public class PortableJestCreateIndexBuilder extends CreateIndex.Builder {
+
+  public static final String INCLUDE_TYPE_NAME_PARAM = "include_type_name";
+
+  private final Version version;
+
+  public PortableJestCreateIndexBuilder(String index, Version version) {
+    super(index);
+    this.version = version;
+  }
+
+  @Override
+  public CreateIndex build() {
+    if (version.equals(Version.ES_V7)) {
+      setParameter(INCLUDE_TYPE_NAME_PARAM, true);
```

Comment:
This seems brittle if/when we add `Version.ES_V8`. Should we instead do something like:
```
if (version.compareTo(Version.ES_V7) >= 0) {
```
This is annoyingly vague and easy to mess up the logic, so we could simplify it to:
```
if (version.isAtLeast(Version.ES_V7)) {
```
if we added a method to the enum:
```
public boolean isAtLeast(Version other) {
  return this.compareTo(other) >= 0;
}
```
WDYT?

Reply:
Hi,
  this parameter is only for ES_V7, it will not work for newer versions. The only solution to have the connector support ES_V8 will be to remove types, what means:

* Code changes to remove usage of types.
* Wait until Jest support ES_V8, or patch Jest Actions Builders to not use types.
* Another option would be to use the Elasticsearch own client.

BTW, removing types is not a backwards compatible change, so we'll need to have a backwards compatible change for the connector build-in.

_Note:_ that patch Jest Action builders to not use types will be quite some work, this is why I created this PR for now and will do the work for ES_V8 in an upcoming PR. 

What do you think?